app/src/main/java/com/android/solvit/shared/model/chat/AiSolverViewModel.kt

```diff
@@ -9,18 +9,42 @@ import com.android.solvit.BuildConfig
 import com.google.ai.client.generativeai.GenerativeModel
 import com.google.ai.client.generativeai.type.BlockThreshold
 import com.google.ai.client.generativeai.type.Content
+import com.google.ai.client.generativeai.type.FunctionType
 import com.google.ai.client.generativeai.type.HarmCategory
 import com.google.ai.client.generativeai.type.SafetySetting
+import com.google.ai.client.generativeai.type.Schema
 import com.google.ai.client.generativeai.type.generationConfig
 import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.StateFlow
 import kotlinx.coroutines.launch
+import kotlinx.serialization.json.Json
 
 class AiSolverViewModel : ViewModel() {
 
   private val _messageContext = MutableStateFlow<List<ChatMessage>>(emptyList())
   val messageContext: StateFlow<List<ChatMessage>> = _messageContext
 
+  // Schema for the AiSolverResponse
+  private val schema =
+      Schema(
+          name = "AiSolverResponse",
+          description = "The Ai response including text and a flag for request creation",
```

Comment:
Nice use of `Schema` to get a structured response from the model. Now we can get a boolean indicating if the user should create a request to solve his problem.